### PR TITLE
giac: 1.4.9 -> 1.5.0

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change
update

this also fixes a bug where the xcas ui (notably menus) were broken because of split outputs. Apparently xcas uses its documentation to populate some ui elements.
I added a few optional dependencies.

###### Things done

* built sage with it. Patch sent upstream. cc @timokau 
* modiffied nauty to also install its libraries. cc @7c6f434c as maintainer

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`) only xcas
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by 7c6f434c — APPROVED** [review]
Looks fine to me. As `giac` is only there for `sage` (and I added it when trying to bump Sage), I would prefer to defer to @timokau — so I will not merge myself earlier than after a couple of says.

File: pkgs/applications/science/math/nauty/default.nix [hunk 1]
```diff
@@ -6,13 +6,20 @@ stdenv.mkDerivation rec {
...
     sha256 = "05z6mk7c31j70md83396cdjmvzzip1hqb88pfszzc6k4gy8h3m2y";
   };
+  outputs = [ "out" "dev" ];
   buildInputs = [];
   installPhase = ''
-    mkdir -p "$out"/{bin,share/doc/nauty}
+    mkdir -p "$out"/{bin,share/doc/nauty} "$dev"/{lib,include/nauty}
 
-    cp $(find . -type f -perm -111 \! -name '*.*' ) "$out/bin"
+    cp $(find . -type f -perm -111 \! -name '*.*' \! -name configure) "$out/bin"
```
Comment (timokau) [inline pkgs/applications/science/math/nauty/default.nix]: Would be better to use `find -exec cp '{}' target \;` instead

File: pkgs/applications/science/math/nauty/default.nix [hunk 2]
```diff
@@ -6,13 +6,20 @@ stdenv.mkDerivation rec {
...
     cp [Rr][Ee][Aa][Dd]* COPYRIGHT This* [Cc]hange* "$out/share/doc/nauty"
+
+    cp *.h $dev/include/nauty
+    for i in *.a; do
+      cp $i $dev/lib/lib$i;
```
Comment (timokau) [inline pkgs/applications/science/math/nauty/default.nix]: Should be quoted

**Review by timokau — CHANGES_REQUESTED** [review]
Thanks for taking care of this! Giac can be a bit of a pain to deal with. Looks good to me, just two minor nitpicks.

**Comment by timokau** [comment]
Also even smaller nitpicks (not necessary to fix those, just my preferences): I think the fixup commit should be merged with the update commit. And if the new dependencies are really optional, it would be nice to create a flag (default enable) for them.

**Review by timokau — APPROVED** [review]

**Comment by timokau** [comment]
Great! I don't think just null'ing the dependencies is a good option, but as I said thats not a blocker. Thanks again!

**Pull request merged**